### ISSUE TITLE
"Trying to read deleted variable" false negative when importing deleted variable

### ISSUE THREAD

**Issue by DetachHead** [python/mypy#13224]
```py
# asdf.py
a = 1
del a

print(a) # error: Trying to read deleted variable
```
```py
# main.py
from asdf import a # no error. runtime ImportError: cannot import name 'a' from 'asdf'

print(a) # no error
```
